src/qt/pivx/masternodewizarddialog.cpp

```diff
@@ -150,45 +150,65 @@ void MasterNodeWizardDialog::accept()
 
 bool MasterNodeWizardDialog::createMN()
 {
-    if (walletModel) {
-        /**
-         *
-        1) generate the mn key.
-        2) create the mn address.
-        3) send a tx with 10k to that address.
-        4) get thereate output.
-        5) use those values on the masternode.conf
-         */
-
-        // First create the mn key
-        CKey secret;
-        secret.MakeNewKey(false);
-        CBitcoinSecret mnKey = CBitcoinSecret(secret);
-        std::string mnKeyString = mnKey.ToString();
-
-        // second create mn address
-        QString addressLabel = ui->lineEditName->text();
-        if (addressLabel.isEmpty()) {
-            returnStr = tr("address label cannot be empty");
-            return false;
-        }
-        std::string alias = addressLabel.toStdString();
+    if (!walletModel) {
+        returnStr = tr("walletModel not set");
+        return false;
+    }
 
-        QString addressStr = ui->lineEditIpAddress->text();
-        QString portStr = ui->lineEditPort->text();
-        if (addressStr.isEmpty() || portStr.isEmpty()) {
-            returnStr = tr("IP or port cannot be empty");
-            return false;
-        }
-        if (!MNModel::validateMNIP(addressStr)) {
-            returnStr = tr("Invalid IP address");
-            return false;
-        }
+    /**
+     *
+    1) generate the mn key.
+    2) create the mn address.
+    3) if there is a valid (unlocked) collateral utxo, use it
+    4) otherwise create a receiving address and send a tx with 10k to it.
+    5) get the collateral output.
+    6) use those values on the masternode.conf
+     */
+
+    // validate IP address
+    QString addressLabel = ui->lineEditName->text();
+    if (addressLabel.isEmpty()) {
+        returnStr = tr("address label cannot be empty");
+        return false;
+    }
+    std::string alias = addressLabel.toStdString();
 
-        // ip + port
-        std::string ipAddress = addressStr.toStdString();
-        std::string port = portStr.toStdString();
+    QString addressStr = ui->lineEditIpAddress->text();
+    QString portStr = ui->lineEditPort->text();
+    if (addressStr.isEmpty() || portStr.isEmpty()) {
+        returnStr = tr("IP or port cannot be empty");
+        return false;
+    }
+    if (!MNModel::validateMNIP(addressStr)) {
+        returnStr = tr("Invalid IP address");
+        return false;
+    }
 
+    // ip + port
+    std::string ipAddress = addressStr.toStdString();
+    std::string port = portStr.toStdString();
+
+    // create the mn key
+    CKey secret;
+    secret.MakeNewKey(false);
+    CBitcoinSecret mnKey = CBitcoinSecret(secret);
+    std::string mnKeyString = mnKey.ToString();
+
+    // Look for a valid collateral utxo
+    COutPoint collateralOut;
+    bool found = false;
+    std::vector<COutPoint> vOuts;
+    walletModel->getPossibleMNCollaterals(vOuts);
+    for (const COutPoint& outPoint : vOuts) {
+        if (!walletModel->isLockedCoin(outPoint.hash, outPoint.n)) {
```

Comment:
wouldn't make more sense to move this check into the `getPossibleMNCollaterals` method and only have here the possible collateral utxo?

Reply:
Yes. The original idea was to have a generic function that returns all "possible" collateral candidates, and filter them outside of it. But since it is used only here, we can make it simpler: look for just *one* possible collateral utxo, save it in `outPoint` and return `found`.